client/src/data_layer_configs/population_by_census_block_config.ts

```diff
@@ -7,46 +7,93 @@ import {
   TileDataLayer,
 } from '@/model/data_layer';
 import { FillLayer } from 'mapbox-gl';
-import { colorMapToBuckets, tileServerHost } from '@/util/data_layer_util';
+import { getLegendBucketsAsList, tileServerHost } from '@/util/data_layer_util';
 
 const DEFAULT_NULL_COLOR = '#d3d3d3';
-const TABLE_NAME = 'public.demographics';
+const ONE_MILLION = 1000000;
+const TEN_THOUSAND = 10000;
+const TABLE_NAME = 'public.demographics_function_source';
 
-/**
- * Maps legend buckets to the hex values.
- */
-const LEGEND_COLOR_MAPPING = [
-  0,
-  '#ECE5FF',
-  800,
-  '#D8CCFF',
-  1200,
-  '#B299FF',
-  1700,
-  '#9E80FF',
-  2200,
-  '#8B66FF',
-  2700,
-  '#774DFF',
+const LEGEND_VALUES_STATES = [
+  {
+    bucketValue: 0,
+    bucketColor: '#D1E4F2',
+  },
+  {
+    bucketValue: ONE_MILLION,
+    bucketColor: '#C7E0F8',
+  },
+  {
+    bucketValue: ONE_MILLION * 2,
+    bucketColor: '#76bbfe',
+  },
+  {
+    bucketValue: ONE_MILLION * 5,
+    bucketColor: '#4786c9',
+  },
+  {
+    bucketValue: ONE_MILLION * 10,
+    bucketColor: '#164E87',
+  },
+  {
+    bucketValue: ONE_MILLION * 20,
+    bucketColor: '#0B2553',
+  },
 ];
 
+// TODO (breuch): automate this. Right now the mathematical relationship
+// between LEGEND_VALUES_STATES and LEGEND_VALUES_COUNTY is not yet clear.
+export const LEGEND_VALUES_COUNTY = [
```

Comment:
thanks for cleaning up these look great!

Reply:
I need to figure out how to generate one legend value from another without having two maps